dataactvalidator/config/sqlrules/c9_award_financial.sql

```diff
@@ -1,27 +1,28 @@
-SELECT afa.row_number, afa.fain, afa.uri
-FROM award_financial_assistance AS afa
-WHERE afa.submission_id = {0}
-    AND (COALESCE(afa.federal_action_obligation, 0) <> 0
+WITH award_financial_assistance_c9_{0} AS
+    (SELECT submission_id,
+		row_number,
+		federal_action_obligation,
+		original_loan_subsidy_cost,
+		fain,
+		uri
+	FROM award_financial_assistance
+	WHERE submission_id = {0})
+SELECT
+    afa.row_number,
+    afa.fain,
+    afa.uri
+FROM award_financial_assistance_c9_{0} AS afa
+WHERE (COALESCE(afa.federal_action_obligation, 0) <> 0
         OR COALESCE(CAST(afa.original_loan_subsidy_cost as numeric),0) <> 0
     ) AND (afa.fain IS NOT NULL
             OR afa.uri IS NOT NULL
-	) AND (afa.row_number NOT IN (
+	) AND afa.row_number NOT IN (
             SELECT afa.row_number
-            FROM award_financial_assistance AS afa
+            FROM award_financial_assistance_c9_{0} AS afa
                 JOIN award_financial AS af
                     ON afa.submission_id = af.submission_id
-                        AND afa.fain = af.fain
-            WHERE afa.submission_id = {0}
-        ) AND afa.row_number NOT IN (
-            SELECT afa.row_number
-            FROM award_financial_assistance AS afa
-                JOIN award_financial AS af
-                    ON afa.submission_id = af.submission_id
-                        AND afa.uri = af.uri
-            WHERE afa.submission_id = {0}
-        )
-    ) AND NOT EXISTS(
-	      SELECT transaction_obligated_amou
-	      FROM award_financial AS af
-	      WHERE transaction_obligated_amou IS NULL
-    );
+                        AND (afa.fain = af.fain
```

Comment:
For `afa.fain = af.fain OR (afa.fain IS NULL AND af.fain IS NULL)`, couldn't that be replaced with `afa.fain IS NOT DISTINCE FROM af.fain`?